File: Data Structures & Algorithms/binary-tree-diameter/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int res = 0;

    int diameterOfBinaryTree(TreeNode* root) {
        dfs(root);
        return res;
    }

    int dfs(TreeNode* curr) {
            if (curr == nullptr) {
                return 0;
            }

            int left = dfs(curr->left);
            int right = dfs(curr->right);
            res = max(res, left + right);
            return 1 + max(left, right);
        }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.count = 0
        maxim = root.val

        def dfs(root, maxim):
            if not root:
                return None
            
            print(root.val)
            if root.val >= maxim:
                self.count += 1
                maxim = root.val
            
            dfs(root.left, maxim)
            dfs(root.right, maxim)
        
        dfs(root, maxim)
        return self.count